test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectWhoIsMakingTheClaimControllerSpec.scala

```diff
@@ -105,201 +111,166 @@ class SelectWhoIsMakingTheClaimControllerSpec
 
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
         }
 
         checkPageIsDisplayed(
-          performAction(
-            Seq.empty
-          ),
+          performAction(Seq("select-who-is-making-the-claim" -> "10"), journeyBindable),
           messageFromMessageKey("select-who-is-making-the-claim.title"),
           doc =>
             doc
               .select(".govuk-error-summary__list > li > a")
               .text() shouldBe messageFromMessageKey(
-              s"select-who-is-making-the-claim.error.required"
+              s"select-who-is-making-the-claim.invalid"
             ),
           BAD_REQUEST
         )
       }
 
-      "an invalid option value is submitted" in {
-
-        def performAction(data: Seq[(String, String)]): Future[Result] =
-          controller.selectDeclarantTypeSubmit()(
+      "the user amends their answer" in forAll(testCases) { (numberOfClaims, journeyBindable) =>
+        def performAction(data: Seq[(String, String)], journeyBindable: JourneyBindable): Future[Result] =
+          controller.changeDeclarantTypeSubmit(journeyBindable)(
             FakeRequest().withFormUrlEncodedBody(data: _*)
           )
 
-        val declarantType = DeclarantType.Importer
-
-        val answers: CompleteDeclarantTypeAnswer = CompleteDeclarantTypeAnswer(declarantType)
+        val declarantType = DeclarantTypeAnswer.Importer
 
-        val draftC285Claim                = sessionWithClaimState(Some(answers))._3
+        val answers                       = declarantType
+        val draftC285Claim                = sessionWithClaimState(Some(answers), Some(numberOfClaims))._3
           .copy(
             declarantTypeAnswer = Some(answers),
             movementReferenceNumber = sampleEntryNumberAnswer()
           )
-        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers))
-
-        val updatedJourney = fillingOutClaim.copy(draftClaim = draftC285Claim)
+        val (session, fillingOutClaim, _) = sessionWithClaimState(Some(answers), Some(numberOfClaims))
+        val updatedJourney                = fillingOutClaim.copy(draftClaim = draftC285Claim)
 
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session.copy(journeyStatus = Some(updatedJourney)))
         }
 
-        checkPageIsDisplayed(
-          performAction(Seq("select-who-is-making-the-claim" -> "10")),
-          messageFromMessageKey("select-who-is-making-the-claim.title"),
-          doc =>
-            doc
-              .select(".govuk-error-summary__list > li > a")
-              .text() shouldBe messageFromMessageKey(
-              s"select-who-is-making-the-claim.invalid"
-            ),
-          BAD_REQUEST
+        checkIsRedirect(
+          performAction(Seq("select-who-is-making-the-claim" -> "0"), journeyBindable),
```

Comment:
Can you define "select-who-is-making-the-claim" as key in the controller and reference this variable in the tests?

Reply:
Wondering is it possible rather than having "0" no matter what it is stands for, to have an item and call index on it?

Reply:
done